cal/cal/static/js/app.js

```diff
@@ -326,11 +337,7 @@ function CategoryListCtrl($scope, $http, CalendarFilterService, CategoryService)
 
   this.hideZeroHoursFilter = function(value, index, array) {
     /* jshint unused:vars */
-    if (this.hideZeroHours && value.hours === 0) {
-      return false;
-    } else {
-      return true;
-    }
+    return !this.hideZeroHours || value.hours !== 0
```

Comment:
alternatively just do return !(this.hideZeroHours && values.hours === 0)? Even tho it's unsimplified, it's still a one-liner and easy enough to understand